synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -192,6 +192,14 @@ async def action_for_event_by_user(
         should increment the unread count, and insert the results into the
         event_push_actions_staging table.
         """
+        if event.internal_metadata.is_historical():
```

Comment:
similar logic is implememted in synapse upstream : https://github.com/matrix-org/synapse/blob/848f7e3d5ff38ca28e56e6143e584974da1eec42/synapse/push/bulk_push_rule_evaluator.py#L334